parsers/NTESMO.py

```diff
@@ -71,90 +75,91 @@ class Generator(TypedDict):
     status_forcelist=[500, 502, 503, 504],
 )
 
-_DT_CLASS = "smp-tiles-article__title"
 
+def _find_link_to_daily_report(target_datetime: datetime, session: Session) -> str:
+    """Scrapes daily report cards to find the link to the data for the target date."""
 
-def construct_latest_index(session: Session) -> dict[date, str]:
-    """Browse all links from the latest daily reports page and index them."""
-    index = {}
-    latest_index_page = session.get(LATEST_URL)
-    soup = BeautifulSoup(latest_index_page.text, "html.parser")
-    for a in soup.find_all("a", href=True):
-        if a["href"].startswith(DATA_DOC_PREFIX):
-            dt = pd.to_datetime(a.find("div", {"class": _DT_CLASS}).text)
-            index[dt.date()] = a["href"]
-    return index
+    today_date = datetime.now(AUSTRALIA_TZ).date()
+    target_datetime = target_datetime.astimezone(AUSTRALIA_TZ)
 
+    # cap target datetime if more recent that what the API makes available:
+    # data is being published after 2 days at the moment
+    _DELAY_IN_DAYS = 2
+    if abs(today_date - target_datetime.date()).days < _DELAY_IN_DAYS:
+        target_datetime = target_datetime - timedelta(days=_DELAY_IN_DAYS)
```

Comment:
thought: should we shift this by `_DELAY_IN_DAYS` + 1 ?

It looks like e.g. today (Sun 21 08:00 UTC / 17:30 ACST ) there is still no data for Fri 19  ACST. So maybe it means that data gets published on the 3rd ACST day?